services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/doi/DoiManagerImpl.java

```diff
@@ -240,10 +242,17 @@ String generateLocationRequestUrl(DoiAssociation association) {
 		final String PERSISTENT_REPOSITORY_ENDPOINT = "https://repo-" + stack + "." + stack + ".sagebase.org/repo/v1";
 		
 		String request = PERSISTENT_REPOSITORY_ENDPOINT + LOCATE_RESOURCE_PATH;
-		
+
+		String encodedId;
+		try {
+			encodedId = URLEncoder.encode(association.getObjectId(), StandardCharsets.UTF_8.toString());
+		} catch (Exception e) {
+			throw new IllegalArgumentException("Failed to encode objectId: " + association.getObjectId(), e);
+		}
```

Comment:
If we don't encode the URL that we register with DataCite, we get an error from Tomcat:

Try for yourself:
https://repo-prod.prod.sagebase.org/repo/v1/doi/locate?portaId=1001&id={%22__type%22:%22STUDY%22,%22Study%22:%22syn52167912%22}&type=PORTAL_RESOURCE